### DISCUSSION TITLE
Display empty DeploymentNode in DeploymentView

### DISCUSSION THREAD

**Discussion by lasagnesmatinmidisoir** [structurizr/ui#68] [Ideas]
Hello,

Is there a way to display an empty DeploymentNode in a Deployment View ?
At the moment, it is not the case due to this line in the function  _addElementInstancesAndDeploymentNodesAndInfrastructureNodes_ in the _DeploymentView.java_ file:

```
        if (hasElementsOrInfrastructureNodes) {
            addElement(deploymentNode, addRelationships);
        }
```

This feature could be useful to display the network on which the rest of the model is deployed.

Best,

**Comment by simonbrowndotje** [comment]
As you've seen, empty deployment nodes are removed from the diagram. I'm not sure I understand what you're trying to achieve though.

**Reply by lasagnesmatinmidisoir** [reply]
Hello !

The idea is that we may want to display parts of the deployment environment without some elements. It could occur that we remove all the content of a deployment node and, although it is empty, its presence in the view is valuable information.

Hope it is clear,
